addons/sale/views/product_views.xml

```diff
@@ -14,7 +14,11 @@
                 <field name="visible_expense_policy" invisible="1"/>
             </field>
             <field name="type" position="after">
-                <field name="invoice_policy" required="1" invisible="not sale_ok"/>
+                <field
+                    name="invoice_policy"
+                    required="1"
+                    invisible="not sale_ok or type == 'combo'"
+                />
```

Comment:
There's an override in `pos_sale` that does the same [1]. Could you remove that override?

[1] https://github.com/odoo/odoo/blob/d0c9ceeb684ac19ff38463ed4abef82420076aac/addons/pos_sale/views/product_views.xml#L9-L11